packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,253 @@
+    endpoint: string,
+  ) {
+    // if token doesn't exist, insert it, on conflict do nothing
+    await this.db.db
+      .insertInto('notification_push_token')
+      .values({
+        did,
+        token,
+        platform,
+        appId,
+        endpoint,
+      })
+      .onConflict((oc) => oc.doNothing())
+      .execute()
+  }
+
+  async getNotificationDisplayAttributes(notif: InsertableNotif) {
+    const {
+      author: authorDid,
+      reason,
+      reasonSubject: subjectUri, // if like/reply/quote/emtion, the post which was liked/replied to/mention is in/or quoted. if custom feed liked, the feed which was liked
+      recordUri,
+    } = notif
+    // 1. Get author's display name
+    const displayName = await this.db.db
+      .selectFrom('profile')
+      .where('creator', '=', authorDid)
+      .select(['displayName'])
+      .executeTakeFirst()
+
+    // if no display name, dont send notification
+    if (!displayName || !displayName?.displayName) {
+      return
+    }
+    const author = displayName.displayName
+
+    // 2. Get post data content
+    // if reply, quote, or mention, get URI of the postRecord
+    // if like, or custom feed like, or repost get the URI of the reasonSubject
+    // if follow, get the URI of the author's profile
+    let title = ''
+    let body = ''
+
+    // check follow first and mention first because they don't have subjectUri and return
+    if (reason === 'follow') {
+      title = 'New follower!'
+      body = `${author} has followed you`
+      return { title, body }
+    } else if (reason === 'mention') {
+      title = `${author} mentioned you`
+      const mentionedPostData = await this.db.db
+        .selectFrom('post')
+        .where('uri', '=', recordUri)
+        .selectAll()
+        .executeTakeFirst()
+      body = mentionedPostData?.text || ''
+      return { title, body }
+    }
+
+    // if no subjectUri, don't send notification
+    // at this point, subjectUri should exist for all the other reasons
+    if (!subjectUri) {
+      return
+    }
+
+    // if no post data, don't send notification
+    const postData = await this.db.db
+      .selectFrom('post')
+      .where('uri', '=', subjectUri)
+      .selectAll()
+      .executeTakeFirst()
+    if (!postData) {
+      return
+    }
+
+    if (reason === 'like') {
+      title = `${author} liked your post`
+      body = postData?.text || ''
+      // custom feed like
+      if (subjectUri?.includes('feed.generator')) {
+        title = `${author} liked your custom feed`
+        body = `${new AtUri(subjectUri).rkey}`
+      }
+    } else if (reason === 'reply') {
+      title = `${author} replied to your post`
+      body = postData?.text || ''
+    } else if (reason === 'quote') {
+      title = `${author} quoted your post`
+      body = postData?.text || ''
+    } else if (reason === 'repost') {
+      title = `${author} reposted your post`
+      body = postData?.text || ''
+
+      // TODO: handle images
```

Comment:
just curious, is this gonna make it in in this PR or save for later?

if not, let's delete the commented coded but can preserve the TODO (but again change to `@TODO`)

Reply:
Nah I don't think it is necessary. I'll remove it for now.